post/invoice.php

```diff
@@ -1428,3 +1453,154 @@
+        $invoice_prefix = sanitizeInput($row['invoice_prefix']);
+        $invoice_number = intval($row['invoice_number']);
+        $invoice_amount = floatval($row['invoice_amount']);
+        $invoice_url_key = sanitizeInput($row['invoice_url_key']);
+        $invoice_balance_query = "SELECT SUM(payment_amount) AS amount_paid FROM payments WHERE payment_invoice_id = $invoice_id";
+        $result_amount_paid = mysqli_query($mysqli, $invoice_balance_query);
+        $row_amount_paid = mysqli_fetch_array($result_amount_paid);
+        $amount_paid = floatval($row_amount_paid['amount_paid']);
+        $invoice_balance = $invoice_amount - $amount_paid;
+
+
+        if ($credit_amount <= 0) {
+            break; // Exit the loop if no credit amount is left
+        }
+
+        if ($invoice_balance <= 0) {
+            continue; // Skip the invoice if it's already paid
+        }
+
+        if ($credit_amount >= $invoice_balance) {
+            $payment_amount = $invoice_balance;
+            $invoice_status = "Paid";
+        } else {
+            $payment_amount = $credit_amount;
+            $invoice_status = "Partial";
+        }
+
+        $invoice_applied_count++;
+        
+        // Subtract the payment amount from the credit amount
+        $credit_amount -= $payment_amount;
+
+        // Get Invoice Remain Balance
+        $remaining_invoice_balance = $invoice_balance - $payment_amount;
+
+        // Add Payment
+        $payment_query = "INSERT INTO payments SET payment_date = CURDATE(), payment_amount = $payment_amount, payment_invoice_id = $invoice_id, payment_account_id = 1, payment_currency_code = '{$credit_row['credit_currency_code']}', payment_reference = 'Credit Applied'";
+        mysqli_query($mysqli, $payment_query);
+        $payment_id = mysqli_insert_id($mysqli);
+
+        // Update Invoice Status
+        $update_invoice_query = "UPDATE invoices SET invoice_status = '{$invoice_status}' WHERE invoice_id = {$invoice_id}";
+        mysqli_query($mysqli, $update_invoice_query);
+
+        // Add Payment to History
+        $history_description = "Payment added";
+        $add_history_query = "INSERT INTO history (history_status, history_description, history_invoice_id) VALUES ('{$invoice_status}', '{$history_description}', {$invoice_id})";
+        mysqli_query($mysqli, $add_history_query);
+
+        // Add to Email Body Invoice Portion
+
+        $email_body_invoices .= "<br>Invoice <a href=\'https://$config_base_url/guest_view_invoice.php?invoice_id=$invoice_id&url_key=$invoice_url_key\'>$invoice_prefix$invoice_number</a> - Outstanding Amount: " . numfmt_format_currency($currency_format, $invoice_balance, $currency_code) . " - Payment Applied: " . numfmt_format_currency($currency_format, $payment_amount, $currency_code) . " - New Balance: " . numfmt_format_currency($currency_format, $remaining_invoice_balance, $currency_code);
+
+    } // End Invoice Loop
+
+
+
+    // Send Email
+    if ($email_receipt == 1) {
```

Comment:
$email_receipt does not appear to be assigned via post or within the apply_credit logic?